web/src/components/stageTemplate/add/Form.js

```diff
@@ -0,0 +1,147 @@
+import PropTypes from 'prop-types';
+import { Formik } from 'formik';
+import { inject, observer } from 'mobx-react';
+import { Row, Col } from 'antd';
+import { toJS } from 'mobx';
+import FormContent from './FormContent';
+import { validateForm } from './validate';
+import { argumentsParamtersField } from '@/lib/const';
+
+@inject('stageTemplate')
+@observer
+export default class StageTemplateForm extends React.Component {
+  constructor(props) {
+    super(props);
+    const {
+      match: { params },
+    } = props;
+    this.update = !!_.get(params, 'templateName');
+    if (this.update) {
+      props.stageTemplate.getTemplate(params.templateName);
+    }
+  }
+
+  static propTypes = {
+    history: PropTypes.object,
+    match: PropTypes.object,
+    stageTemplate: PropTypes.object,
+    initialFormData: PropTypes.object,
+    setTouched: PropTypes.func,
+    isValid: PropTypes.bool,
+    values: PropTypes.object,
+  };
+
+  handleCancle = () => {
+    const { history } = this.props;
+    history.push('/stageTemplate');
+  };
+
+  mapRequestFormToInitForm = data => {
+    const alias = _.get(
+      data,
+      ['metadata', 'annotations', 'cyclone.dev/alias'],
+      ''
+    );
+    const description = _.get(
+      data,
+      ['metadata', 'annotations', 'cyclone.dev/description'],
+      ''
+    );
+    const spec = this.generateSpecObj(data);
+    return {
+      metadata: { alias, description },
+      spec,
+    };
+  };
+
+  generateSpecObj = data => {
+    const specData = _.get(data, 'spec', {});
+    let defaultSpec = {
+      pod: {
+        inputs: {
+          arguments: argumentsParamtersField,
+          resources: [],
+        },
+        outputs: {
+          resources: [],
+        },
+        spec: {
+          containers: [
+            {
+              env: [],
+              image: '{{ image }}',
+              args: ['/bin/sh", "-e", "-c", "{{{ cmd }}}'],
+            },
+          ],
+        },
+      },
+    };
+    return _.assign(defaultSpec, specData);
+  };
+
+  initFormValue = () => {
+    const templateInfo = toJS(this.props.stageTemplate.template);
+    return this.mapRequestFormToInitForm(templateInfo);
```

Comment:
map? or merge?

Reply:
map ,this map is meaning just like source map , Map form data and real data